### PR TITLE
tabix.Index.Chunks now expects a record interface

### PR DESCRIPTION
**Pull request by brentp**

### CODE REVIEW

File: tabix/tabix.go [hunk 1]
```diff
@@ -98,8 +97,13 @@ func (i *Index) Add(r Record, c bgzf.Chunk, placed, mapped bool) error {
 }
 
 // Chunks returns a []bgzf.Chunk that corresponds to the given genomic interval.
-func (i *Index) Chunks(r *sam.Reference, beg, end int) []bgzf.Chunk {
-	chunks := i.idx.Chunks(r.ID(), beg, end)
+func (i *Index) Chunks(r Record) []bgzf.Chunk {
+	refName := r.RefName()
+	id, ok := i.nameMap[refName]
+	if !ok {
+		return make([]bgzf.Chunk, 0)
```
Comment (kortschak) [inline tabix/tabix.go]: This should return nil.

**Comment by kortschak** [comment]
While we are breaking API, how do you feel about adding an error return to this? Currently we return an empty []bgzf.Chunk, but there are two possible reasons for this - the reference is not known or the interval is outside the range of the reference. It would mean that Chunks can't be used inline, but it is probably worth doing. I'll add this if you think it's worthwhile - it touches more than just here.

**Comment by brentp** [comment]
that makes sense to me.

**Comment by kortschak** [comment]
OK, I've got that change ready. I just need the nit fixed and I can move on that.

**Comment by brentp** [comment]
ok. done. I just sqaushed and force pushed.

**Comment by kortschak** [comment]
LTGM, thanks.

**Pull request merged**